pkg/kubectl/cmd/config/create_cluster.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"errors"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+	clientcmdapi "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/api"
+)
+
+type createClusterOptions struct {
+	pathOptions           *pathOptions
+	name                  string
+	server                string
+	apiVersion            string
+	insecureSkipTLSVerify bool
+	certificateAuthority  string
+}
+
+func NewCmdConfigSetCluster(out io.Writer, pathOptions *pathOptions) *cobra.Command {
+	options := &createClusterOptions{pathOptions: pathOptions}
+
+	cmd := &cobra.Command{
+		Use:   "set-cluster name [server] [insecure-skip-tls-verify] [certificate-authority] [api-version]",
```

Comment:
set-credentials lgtm. Gcloud uses "auth login", but the usage is different (opens a browser window to authorize a service account).